SportsHub/SportsHub.DAL/Data/Configurations/UserEntityTypeConfiguration.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using SportsHub.Domain.Models;
+
+namespace SportsHub.DAL.Data.Configurations
+{
+    public class UserEntityTypeConfiguration : IEntityTypeConfiguration<User>
+    {
+        public void Configure(EntityTypeBuilder<User> user)
+        {
+            user.Property(x => x.Email)
+                .IsRequired(true)
+                .HasMaxLength(50)
+                .IsUnicode(false);
+
+            user.Property(x => x.Username)
+                .IsRequired(true)
```

Comment:
I think that UserName must be Unique

Reply:
Can ```Username``` be made unique this way:
```CSharp
user.HasAlternateKey(x => x.Username).HasName("AlternateKey_Username");
```

Reply:
better to use IsUnique() there. I suppose that name may be changed, but with AlternateKey in EF you are not able to do it, also everywhere we use ID as a foreign key, so there no needs in alternate key

Reply:
There is no IsUnique() for me:
![image](https://user-images.githubusercontent.com/44908454/190344689-6efb2d26-fcf5-440d-9ec9-63d0de290a04.png)
The best solution I found on StackOverflow is this one: https://stackoverflow.com/questions/4413084/unique-constraint-in-entity-framework-code-first
Should we go with it?

Reply:
user .HasIndex(x => new {  x.UserName})
   .IsUnique()
try this one, please

Reply:
This works, thanks!

Reply:
Implemented